src/Lykke.AzureStorage/Tables/Decorators/ExplicitAppInsightsAzureTableStorageDecorator.cs

```diff
@@ -0,0 +1,262 @@
+            => WrapAsync(() => _impl.GetDataByChunksAsync(partitionKey, chunks), nameof(GetDataByChunksAsync));
+
+        public Task ScanDataAsync(string partitionKey, Func<IEnumerable<TEntity>, Task> chunk)
+            => WrapAsync(() => _impl.ScanDataAsync(partitionKey, chunk), nameof(ScanDataAsync));
+
+        public Task ScanDataAsync(TableQuery<TEntity> rangeQuery, Func<IEnumerable<TEntity>, Task> chunk)
+            => WrapAsync(() => _impl.ScanDataAsync(rangeQuery, chunk), nameof(ScanDataAsync));
+
+        public Task<TEntity> FirstOrNullViaScanAsync(string partitionKey, Func<IEnumerable<TEntity>, TEntity> dataToSearch)
+            => WrapAsync(() => _impl.FirstOrNullViaScanAsync(partitionKey, dataToSearch), nameof(FirstOrNullViaScanAsync));
+
+        public Task<IEnumerable<TEntity>> GetDataAsync(string partition, Func<TEntity, bool> filter = null)
+            => WrapAsync(() => _impl.GetDataAsync(partition, filter), nameof(GetDataAsync));
+
+        public Task<TEntity> GetTopRecordAsync(string partition)
+            => WrapAsync(() => _impl.GetTopRecordAsync(partition), nameof(GetTopRecordAsync));
+
+        public Task<IEnumerable<TEntity>> GetTopRecordsAsync(string partition, int n)
+            => WrapAsync(() => _impl.GetTopRecordsAsync(partition, n), nameof(GetTopRecordsAsync));
+
+        public Task<IEnumerable<TEntity>> GetDataRowKeysOnlyAsync(IEnumerable<string> rowKeys)
+            => WrapAsync(() => _impl.GetDataRowKeysOnlyAsync(rowKeys), nameof(GetDataRowKeysOnlyAsync));
+
+        public Task<IEnumerable<TEntity>> WhereAsyncc(TableQuery<TEntity> rangeQuery, Func<TEntity, Task<bool>> filter = null)
+            => WrapAsync(() => _impl.WhereAsyncc(rangeQuery, filter), nameof(WhereAsyncc));
+
+        public Task<IEnumerable<TEntity>> WhereAsync(TableQuery<TEntity> rangeQuery, Func<TEntity, bool> filter = null)
+            => WrapAsync(() => _impl.WhereAsync(rangeQuery, filter), nameof(WhereAsync));
+
+        public Task ExecuteAsync(TableQuery<TEntity> rangeQuery, Action<IEnumerable<TEntity>> yieldResult, Func<bool> stopCondition = null)
+            => WrapAsync(() => _impl.ExecuteAsync(rangeQuery, yieldResult, stopCondition), nameof(ExecuteAsync));
+
+        public Task DoBatchAsync(TableBatchOperation batch)
+            => WrapAsync(() => _impl.DoBatchAsync(batch), nameof(DoBatchAsync));
+
+        public Task<IPagedResult<TEntity>> ExecuteQueryWithPaginationAsync(TableQuery<TEntity> query, PagingInfo pagingInfo)
+            => WrapAsync(() => _impl.ExecuteQueryWithPaginationAsync(query, pagingInfo), nameof(ExecuteQueryWithPaginationAsync));
+
+        public Task CreateTableIfNotExistsAsync()
+            => WrapAsync(() => _impl.CreateTableIfNotExistsAsync(), nameof(CreateTableIfNotExistsAsync));
+
+        #endregion
+
+        #region decoration logic
+
+        private async Task<TResult> WrapAsync<TResult>(Func<Task<TResult>> func, string callName)
```

Comment:
These Wrap methods can be moved to the separate AppInsightTracer class, or something like that. And this class could be used in the Table, Blob and Queue storage without copy-pasting Wrap methods